Update the Readme for the lib-megashark (#19)
Scille/megashark-lib: README.md

# Generic UI library based on Ionic

> The megashark-lib library is a UI component library. It is used to centralise all the elements that will be reused throughout the various Parsec projects (images, typography, UI components, global translation).

Reviewer: ```suggestion
The megashark-lib library is a UI component library. It is used to centralize all the elements that will be reused throughout the various Parsec projects (images, typography, UI components, global translation).
```

## 🏁 Installation

### Clone the local directory

You will need to clone the library directory locally using git.

```
git clone git@github.com:Scille/megashark-lib.git
```

## 📁 Folder names

**src** (local app): folder used to test the application locally. It includes a component display page.

> **lib**: includes all the libraryce components that you want to have in common in the library.

Reviewer: `libraryce` ?

## 🧑‍💻 Use in the development space

### Modify a megashark-lib component and see the result

> > In the **megashark-lib**:

Reviewer: ```suggestion
> In **megashark-lib**:
```

```
npm run web:open
```

When you push on `master` → This creates a new version of the library.

### Modify a component of **megashark-lib** at the same time as using it in **parsec-cloud**

This allows you to see changes made to the library and see them displayed in real time in the `parsec-cloud` project.  Please follow all the commands in the order below.

1️⃣ **Build the project and install it globally**

> > In the **megashark-lib**

Reviewer: ```suggestion
> In **megashark-lib**
```

```
npm run build:local
```

The command includes (not to be written in the terminal) :

`npm link`: to create a link to another project from a local environment.
`vite build --watch`: to rebuild automatically as soon as there is a modification.

2️⃣ **Link the libraries from parsec-cloud**.

> > In the **parsec-cloud**

Reviewer: ```suggestion
> In **parsec-cloud**
```

```
npm run lib:local:start
```

The command includes (not to be written in the terminal):
`npm link lib-megashark`: allows you to link the lib-megashark library (installed from the machine to github)

3️⃣ **Start development**

> > In the **parsec-cloud**

Reviewer: ```suggestion
> In **parsec-cloud**
```

```
npm run web:open //web
npm run electron:open //electron
```

4️⃣ **Reset installation to state (with remote project)**

⚠️ Required to keep the project with the remote link.

> > In the **parsec-cloud**

Reviewer: ```suggestion
> In **parsec-cloud**
```

# Generic UI library based on Ionic

> The megashark-lib library is a UI component library. It is used to centralize all the elements that will be reused throughout the various Parsec projects (images, typography, UI components, global translation).

Reviewer 2: ```suggestion
The megashark-lib is a UI component library. It is used to centralize all the elements that will be reused throughout the various Parsec projects (images, typography, UI components, global translation).
```

## 🏁 Installation

### Clone the local directory

You will need to clone the library directory locally using git.

```
git clone git@github.com:Scille/megashark-lib.git

> ```

Reviewer 2: Installation needs npm install too

## 📁 Folder names

**src** (local app): folder used to test the application locally. It includes a component display page.

**lib**: includes all the library components that you want to have in common in the library.

## 🧑‍💻 Use in the development space

### Modify a megashark-lib component and see the result

> In **megashark-lib**:

```
npm run web:open
```

> When you push on `master` → This creates a new version of the library.

Reviewer 3: Not really, resyncs should not be done all the time, they should be planned, with a tag and everything. The process of creating a tag (or a "new version") does not need to be detailled, it's up to the maintainer.

### Modify a component of **megashark-lib** at the same time as using it in **parsec-cloud**

> This allows you to see changes made to the library and see them displayed in real time in the `parsec-cloud` project. Please follow all the commands in the order below.

Reviewer 2: ```suggestion
This allows you to see changes made to the library and see them displayed in real time in the project using this library. Please follow all the commands in the order below.
```

1️⃣ **Build the project and install it globally**

> In **megashark-lib**

```
npm run build:local
```

The command includes (not to be written in the terminal) :

`npm link`: to create a link to another project from a local environment.
`vite build --watch`: to rebuild automatically as soon as there is a modification.

2️⃣ **Link the libraries from parsec-cloud**.

> In **parsec-cloud**

```
npm run lib:local:start
```

The command includes (not to be written in the terminal):
`npm link lib-megashark`: allows you to link the lib-megashark library (installed from the machine to github)

3️⃣ **Start development**

> In **parsec-cloud**

```

> npm run web:open //web

Reviewer 3: Comments use `#`, not `//`

npm run electron:open //electron
```

4️⃣ **Reset installation to state (with remote project)**

⚠️ Required to keep the project with the remote link.

> In **parsec-cloud**

```

> npm run lib:local:stop

Reviewer 3: Renamed to `megashark:local:stop`

## Use the library in your project

> In **your project** and precisely in the `package.json` file:

In order to create the link between the project and the megashark-lib, you should add the link.

`git`+ `https://github/...`

> Example:

```
"megashark-lib": "git+https://github.com/Company/megashark-lib.git",
```

## Modify a megashark-lib component and see the result

> In **megashark-lib**:

```
npm run web:open
```

## Make modifications in **megashark-lib** and update a local project using the library at the same time

This allows you to see changes made to the library and see them displayed in real time in the project using this library. Please follow all the commands in the order below.

### 1️⃣ Build the project and install it globally

> In **megashark-lib**

```
npm run build:local
```

The command includes (not to be written in the terminal) :

`npm link`: to create a link to another project from a local environment.
`vite build --watch`: to rebuild automatically as soon as there is a modification.

### 2️⃣ Link the libraries from your project

> In **your project**

```
npm run megashark:local:start
```

The command includes (not to be written in the terminal):

> `npm link lib-megashark`: allows you to link the lib-megashark library (installed from the machine to github)

Reviewer 2: ```suggestion
`npm link megashark-lib`: allows you to link the megashark-lib library (installed from the machine to github)
```

### 3️⃣ Start development

> In **your project**

```
# For web project

> npm run web:open#web

Reviewer 2: ```suggestion
npm run web:open
```
